### PR TITLE
Fixed num_channels!=3 normalization training

### LINKED ISSUE #20580: VideoMAE with `num_channels!=3` needs a small fix
## Issue

The VideoMAE model doesn't work for non-RGB videos (when `num_channels!=3`). I believe this is caused by the harcoded image-net means and stds in the following lines:

https://github.com/huggingface/transformers/blob/d51e7c7e8265d69db506828dce77eb4ef9b72157/src/transformers/models/videomae/modeling_videomae.py#L824L826

## Code to Reproduce

```
import torch
from transformers import VideoMAEConfig, VideoMAEForPreTraining

NUM_CHANNELS = 1

config = VideoMAEConfig(num_channels=NUM_CHANNELS)
model = VideoMAEForPreTraining(config)

pixel_values = torch.rand(1, 16, NUM_CHANNELS, 224, 224)

num_patches_per_frame = (model.config.image_size // model.config.patch_size) ** 2
seq_length = (model.config.num_frames // model.config.tubelet_size) * num_patches_per_frame
bool_masked_pos = torch.randint(0, 2, (1, seq_length)).bool()

outputs = model(pixel_values, bool_masked_pos=bool_masked_pos)
loss = outputs.loss
```
This produces the following error: ```File ".../lib/python3.10/site-packages/torch/functional.py", line 74, in broadcast_tensors
    return _VF.broadcast_tensors(tensors)  # type: ignore[attr-defined]
RuntimeError: The size of tensor a (512) must match the size of tensor b (1536) at non-singleton dimension 2```

Setting NUM_CHANNELS = 3 works fine.

## Potential Fix

Since we don't have a `_DEFAULT_MEAN/STD` for non-RGB images, we can just replace `frames=pixel_values` and disallow `norm_pix_loss=False` if `num_channels!=3`. With this modification, the code works on my machine. I am willing to fix and submit a PR.

### PR DESCRIPTION
**Pull request by layjain**
Fixes #20580 and #19913 

## Who can review?

@NielsRogge

### CODE REVIEW

**Review by NielsRogge — APPROVED** [review]
Thanks for improving this!

Could you make sure the CI is green (all checks passing) such that we can merge your PR?

You can click on each check => "artifacts" => "failures long" to see why a check failed

**Comment by sgugger** [comment]
Not exactly, the issue with your CircleCI permissions, the tests won't run.

Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

**Comment by layjain** [comment]
I have been now stuck on this for a while. I refreshed the permissions and re-ran the CircleCI checks, and I get the error:
"Resource class docker for xlarge is not available for your project, or is not a valid resource class. This message will often appear if the pricing plan for this project does not support docker use."

**Comment by sgugger** [comment]
You might need to push an empty commit to re-trigger the tests after refreshing your permissions.

**Comment by NielsRogge** [comment]
Hi @layjain let me know whether you could pick this up :)

**Comment by ydshieh** [comment]
FYI I pushed an empty commit to trigger CI

**Comment by layjain** [comment]
Hi @NielsRogge , I have fixed the CircleCI permissions, can this be merged.

**Comment by sgugger** [comment]
@layjain The CI is currently running under your profile and not the Hugging Face profile, and as such our tests are mostly not run (there should be 22 checks here). If you rebase your PR on main you will see a new check failing (we added a fix to detect this recently).

File: src/transformers/models/videomae/modeling_videomae.py [hunk 1]
```diff
@@ -859,6 +863,10 @@ def forward(
                     tubelet_size * patch_size * patch_size * num_channels,
                 )
             else:
+                if self.config.num_channels != 3:
+                    raise ValueError(
+                        "Can't unnormalize non-RGB images. Consider setting config.norm_pix_loss to False."
```
Comment (ydshieh) [inline src/transformers/models/videomae/modeling_videomae.py:868]: I feel a bit confused here. We are in  the `else` block, which means we are already `self.config.norm_pix_loss == False`..
But the message says `Consider setting config.norm_pix_loss to False. ...`. Also this block doesn't have normalization, and the block  `if self.config.norm_pix_loss:` has `# step 4: normalize`.

@NielsRogge  @layjain  Could you double check?

**Review by sgugger — APPROVED** [review]

**Pull request merged**